lbry/dht/protocol/iterative_find.py

```diff
@@ -217,38 +233,74 @@ def callback(_):
         self.running_probes[peer] = t
 
     def _log_state(self):
-        log.debug("[%s] check result: %i active nodes %i contacted",
-                  self.key.hex()[:8], len(self.active), len(self.contacted))
+        log.debug("%s[%x] [%s] check result: %i active nodes %i contacted %i produced %i queued",
+                  type(self).__name__, id(self), self.key.hex()[:8],
+                  len(self.active), len(self.contacted),
+                  self.iteration_count, self.iteration_queue.qsize())
+
+    async def _generator_func(self):
```

Comment:
Is it necessary to separate the generator when using the `aclosing` for fixing this bug? I tried on master and it worked without doing it.
I see there are some nice changes in behavior which would be cool to have in a new PR with the others. This helps testing and tidying the changelog.